File: ios/App/App/Features/Components/BrandWordmark.swift
import SwiftUI

extension Color {
  /// Pora indigo (`#8B7BFF`) — the one chromatic brand accent, per
  /// `branding/BRAND.md` §6. Used for the wordmark dot and brand highlights.
  static let poracodeIndigo = Color(red: 0.545, green: 0.486, blue: 1.0)
}

/// The `Pora·code` brand wordmark: "Pora" in bold, the indigo baseline dot, then
/// "code" in semibold (`branding/BRAND.md` §5). Mirrors the web mark in
/// `src/renderer/components/common/BrandWordmark.tsx`.
///
/// Two details the brand guide is strict about:
///
/// - **The dot is a drawn `Circle`, never a typed period.** The Pora dot is a
///   true round circle, and geometric sans faces (Geist, and SF at small optical
///   sizes) draw "." as a square — a typed period would be the wrong mark.
/// - **It sits at the baseline**, not centered. Positioning comes from
///   `firstTextBaseline` alignment rather than a fixed offset, and the dot's
///   diameter, side gaps, and baseline overshoot are `@ScaledMetric` fractions of
///   the text style's own point size (the same em ratios the web SVG uses). So
///   the lockup holds its proportions and stays aligned at every Dynamic Type
///   size, accessibility sizes included.
///
/// To assistive tech this is a single word: "Poracode" — the written product
/// name. The dot is a logo device and is not spoken.
struct BrandWordmark: View {
  private let textStyle: Font.TextStyle
  private let dotColor: Color

  // Ratios of the font's point size. Dot diameter and baseline overshoot match the
  // web mark's SVG geometry (0.18em wide, sunk 0.05em below the baseline). The side
  // gaps are tightened from the web's 0.13/0.06: SF carries more side bearing than
  // Geist, so the web values read as a word space before the dot and as a collision
  // after it. These are the values that reproduce the web mark's *apparent* spacing.
  @ScaledMetric private var dotDiameter: CGFloat
  @ScaledMetric private var gapBeforeDot: CGFloat
  @ScaledMetric private var gapAfterDot: CGFloat
  @ScaledMetric private var baselineOvershoot: CGFloat

  init(textStyle: Font.TextStyle = .title2, dotColor: Color = .poracodeIndigo) {
    self.textStyle = textStyle
    self.dotColor = dotColor
    let size = Self.nominalPointSize(for: textStyle)
    _dotDiameter = ScaledMetric(wrappedValue: size * 0.18, relativeTo: textStyle)
    _gapBeforeDot = ScaledMetric(wrappedValue: size * 0.07, relativeTo: textStyle)
    _gapAfterDot = ScaledMetric(wrappedValue: size * 0.05, relativeTo: textStyle)
    _baselineOvershoot = ScaledMetric(wrappedValue: size * 0.05, relativeTo: textStyle)
  }

  var body: some View {
    HStack(alignment: .firstTextBaseline, spacing: 0) {
      Text(verbatim: "Pora")
        .font(.system(textStyle, weight: .bold))
      dot
      Text(verbatim: "code")
        .font(.system(textStyle, weight: .semibold))
    }
    .accessibilityElement(children: .ignore)
    .accessibilityLabel(Text(verbatim: "Poracode"))
  }

  private var dot: some View {
    // Capture the actor-isolated scaled metric before entering SwiftUI's
    // Sendable alignment closure (Swift 6 diagnoses direct property access).
    let overshoot = baselineOvershoot
    return Circle()
      .fill(dotColor)
      .frame(width: dotDiameter, height: dotDiameter)
      .padding(.leading, gapBeforeDot)
      .padding(.trailing, gapAfterDot)
      // Align a point just above the circle's bottom edge with the text
      // baseline, so the round dot overshoots it the way round glyphs do.
      .alignmentGuide(.firstTextBaseline) { $0[.bottom] - overshoot }
  }

  /// Default point size of a text style at the `.large` content size, used as the
  /// base the `@ScaledMetric` ratios scale from.
  private static func nominalPointSize(for style: Font.TextStyle) -> CGFloat {
    switch style {
    case .largeTitle: return 34
    case .title: return 28
    case .title2: return 22
    case .title3: return 20
    case .headline, .body: return 17
    case .callout: return 16
    case .subheadline: return 15
    case .footnote: return 13
    case .caption: return 12
    case .caption2: return 11
    default: return 17
    }
  }
}

/// In-app continuation of the platform launch screen while durable session state
/// is being read. Network reconnects never remain here: once a stored profile is
/// known, the root presents Home and lets recovery continue behind it.
struct BrandLaunchView: View {
  var body: some View {
    ZStack {
      Color(.systemBackground)
        .ignoresSafeArea()
      BrandWordmark(textStyle: .largeTitle)
    }
  }
}

#Preview("Brand wordmark") {
  VStack(alignment: .leading, spacing: 20) {
    BrandWordmark(textStyle: .largeTitle)
    BrandWordmark(textStyle: .title)
    BrandWordmark()
    BrandWordmark(textStyle: .body)
    BrandWordmark(textStyle: .footnote)
  }
  .padding()
}

File: android/app/src/main/kotlin/com/poracode/app/ui/components/BrandWordmark.kt
package com.poracode.app.ui.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.text.InlineTextContent
import androidx.compose.foundation.text.appendInlineContent
import androidx.compose.material3.LocalTextStyle
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.clearAndSetSemantics
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.heading
import androidx.compose.ui.text.Placeholder
import androidx.compose.ui.text.PlaceholderVerticalAlign
import androidx.compose.ui.text.SpanStyle
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.buildAnnotatedString
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.unit.em

/** Accessible name of the whole mark — it reads as one product name, not three parts. */
private const val WORDMARK_LABEL = "Poracode"
private const val DOT_SLOT = "pora-dot"

/**
 * The Pora·code brand wordmark — bold "Pora", the accent dot on the baseline, and
 * semibold "code". Mirrors the web `BrandWordmark` and the brand guide.
 *
 * The dot is a **drawn circle**, never a typed period: the geometric sans faces the
 * apps ship render "." as a square, which is wrong for this mark. It is laid out as
 * inline text content aligned `AboveBaseline`, so its bottom edge sits exactly on
 * the text baseline at every font size and font scale.
 *
 * The parts are hidden from assistive tech and the whole mark exposes a single
 * "Poracode" description.
 */
@Composable
fun BrandWordmark(
    modifier: Modifier = Modifier,
    style: TextStyle = LocalTextStyle.current,
    color: Color = Color.Unspecified,
    dotColor: Color = MaterialTheme.colorScheme.primary,
    isHeading: Boolean = false,
) {
    val mark = buildAnnotatedString {
        withStyle(SpanStyle(fontWeight = FontWeight.Bold)) { append("Pora") }
        appendInlineContent(DOT_SLOT, "·")
        withStyle(SpanStyle(fontWeight = FontWeight.SemiBold)) { append("code") }
    }
    val dot = mapOf(
        DOT_SLOT to InlineTextContent(
            Placeholder(
                // Diameter plus the optical side bearing the mark needs on each side.
                width = 0.34.em,
                height = 0.2.em,
                placeholderVerticalAlign = PlaceholderVerticalAlign.AboveBaseline,
            ),
        ) {
            Canvas(Modifier.fillMaxSize()) {
                val radius = size.height / 2f
                drawCircle(
                    color = dotColor,
                    radius = radius,
                    center = Offset(size.width / 2f, size.height - radius),
                )
            }
        },
    )
    Text(
        text = mark,
        inlineContent = dot,
        style = style,
        color = color,
        textAlign = TextAlign.Center,
        modifier = modifier.clearAndSetSemantics {
            contentDescription = WORDMARK_LABEL
            if (isHeading) heading()
        },
    )
}
